configure.d

```diff
@@ -0,0 +1,653 @@
+        /**
+         * The name of the file this path represents.
+         *
+         * This is a name for the file that is used in error messages.
+         */
+        string name;
+
+        /**
+         * A set of standard paths to which to search for the file this path
+         * represents.
+         */
+        const(string)[] standardPaths;
+
+        /**
+         * The custom path that was specified when invoking this configuration
+         * script, or `null` if no custom path was specified.
+         */
+        string specifiedPath;
+
+        /// The actual file to look for in `standardPaths` and `specifiedPath`.
+        string fileToCheck;
+
+        /// Local cache for the full path to the file.
+        string path_;
+    }
+
+    alias path this;
+
+    /**
+     * Constructs a new instance of this class.
+     *
+     * Params:
+     *  name = the name of the file this path represents
+     *
+     *  standardPaths = a set of standard paths to which to search for the file
+     *      this path represents
+     *
+     *  specifiedPath = the custom path that was specified when invoking this
+     *      configuration script, or `null` if no custom path was specified
+     *
+     *  fileToCheck = the actual file to look for in `standardPaths` and
+     *      `specifiedPath`
+     */
+    this(string name, const(string)[] standardPaths,
+        string specifiedPath, string fileToCheck)
+    {
+        this.name = name;
+        this.standardPaths = standardPaths;
+        this.specifiedPath = specifiedPath;
+        this.fileToCheck = fileToCheck;
+    }
+
+    /**
+     * Returns the full path to the file this path represents as a string.
+     *
+     * If `specifiedPath` is non-empty, `fileToCheck` will be searched for in
+     * `specifiedPath`. Otherwise `fileToCheck` will be searched for in
+     * `standardPaths`.
+     *
+     * Returns: the full path to the file this path represents
+     */
+    string path()
+    {
+        if (path_.ptr)
+            return path_;
+
+        return path_ = specifiedPath.empty ? standardPath : customPath;
+    }
+
+    override string toString()
+    {
+        return path;
+    }
+
+protected:
+
+    /**
+     * Returns what kind of file this path points to, i.e. library or
+     * executable.
+     *
+     * This is used in error messages.
+     *
+     * Returns: what kind of file this path points
+     */
+    abstract string type();
+
+
+    /**
+     * Returns the full path of `fileToCheck` by searching in `standardPaths`.
+     *
+     * Returns: the full path of `fileToCheck` by searching in `standardPaths`
+     *
+     * Throws: an `Exception` if `fileToCheck` cannot be found in any of the
+     *  paths in `standardPath`
+     */
+    string standardPath()
+    {
+        auto errorMessage = format("Could not find %s in any of the standard " ~
+            "paths for %s: \n%s\nPlease specify a path manually using " ~
+            "'./configure --%s-path=<path>'.",
```

Comment:
This prints `Please specify a path manually using './configure --llvm-path=<path>'` which is wrong (correct flag is `--llvm-lib-path`)